x-pack/solutions/security/plugins/security_solution/docs/adrs/threat_hunting/0002_timeline_ui_batching_strategy.md

```diff
@@ -0,0 +1,172 @@
+- **id**: Timeline Id
+- **fields**:  List of requested fields
+- **filterQuery**:  combined query consisting: 
+  * **KQL/Lucene Query**
+  * **Filters**
+  * **Data Providers**
+- **runtimeMappings**: List of runtime mappings that need to be fetched
+- **language**: KQL | Lucene
+- **limit**: Sample Size or Batch Size
+- **sort**
+
+## Batch fetch/re-fetch Strategy
+Batch fetch/re-fetch strategy depends on the scenario or the action the user is taking. This section will lay out all the scenarios in which a batch should be re-fetched and its corresponding strategy.
+
+### New Batch Requested
+When a user requests a new batch after clicking on `Load more` as shown below, a new batch will be fetched. For example, if user is on Batch 0 and clicks on Load more, a new batch will be fetched with below strategy :
+
+![](./images/load_more_timeline.png)
+
+```js
+{
+  /* Fetch Batch 1 */
+  activePage: 1, /* current batch */
+  querySize: 500 /* batch/sample size */
+}
+```
+
+Consequently, if a user requests a new batch again ( for the 3rd time ), Batch 2 will be fetched with below strategy.
+
+```js
+{
+	/* Fetch Batch 2 */
+	activePage: 2,
+	querySize: 500 /* batch/sample size */
+}
+```
+ 
+### Refresh Button
+
+When a user clicks on the Refresh button in the query tab, it should explicitly re-fetch new data ir-respective of the fact if any batch parameters have been changed or not.
+
+#### When user is on the first (0th) batch
+When user is on the first batch i.e. Batch 0, it will be re-fetched with the below strategy.
+
+  ```js
+  {
+    /* Fetch Batch 0 */
+    activePage: 0,
+    querySize: 500 /* batch/sample size */
+  }
+  ```
+
+#### When user has already fetched multiple batches
+Let’s say the user has already fetched 3 batches and currently the user is on Batch 2. Clicking on Refetch will reset the result set to 0th batch:
+
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+### Adding a new column
+Adding a new column should not affect the result set that is currently being displayed in front of the user. We should fetch a complete result set ir-respective of the batch user is on.
+
+#### When user is on the first(0th) batch
+When user is on the first batch i.e. Batch 0, it will be re-fetched with the below strategy.
+
+```js
+{
+	/* Fetch Batch 0 */
+	activePage: 0,
+	querySize: 500 /* batch/sample size */
+}
+```
+
+#### When user has already fetched multiple batches
+
+When user is on the 4th batch i.e. Batch 3, it will be re-fetched with the below strategy. 
+
+```js
+{
+  /* Fetch Batch 0 */
+  activePage: 0,
+  querySize: 2000 /* cumulative 4 batches ( 500 * 4 ) */
+}
+```
+
+One important thing to notice here is that the querySize is 2000 instead of 500 which means we are cumulatively for all the visible batches. This comes with its own performance considerations but since the maximum allowed resultset is 10K, I think we can work through it.
```

Comment:
I think these 2 sections could also be combined, saying that we fetch all the batches, with an example that could be

```js
{
	/* Fetch all Batches */
	activePage: 0,
	querySize: x * 500 /* x the number of batches already fetched */
}
```

Reply:
They could be but i wanted to be explicit in the documentation. Otherwise, i thought that there will be no difference between documentation and code.